src/app/core/library-module/library.service.ts

```diff
@@ -63,7 +65,14 @@ export class LibraryService {
       )
       .toPromise()
       .then((val: any) => {
-        this.libraryItems = val.userLibraryItems.map(object => new LearningObject(object.learningObject));
+        this.libraryItems = val.userLibraryItems
+          .filter(object => object && object.learningObject) // Filter out objects without learningObject
+          .map(object => {
+            if (object.learningObject) {
+              object.learningObject.id = object.learningObject._id;
+            }
+            return new LearningObject(object.learningObject);
+          });
```

Comment:
Based on the staging data, this object would not be undefined. Lets pull staging data to ensure all values are being returned. Otherwise, there's a bigger issue we need to address.

![image](https://github.com/user-attachments/assets/0c562f5c-3ad8-4765-95fb-230ace0db14a)

Reply:
Out of scope. Totally different bug that I'll create another story for